src/android/CameraLauncher.java

```diff
@@ -498,15 +498,19 @@ private void refreshGallery(Uri contentUri)
 
 
 private String ouputModifiedBitmap(Bitmap bitmap, Uri uri) throws IOException {
+        // Some content: URIs do not map to file paths (e.g. picasa).
+        String realPath = FileHelper.getRealPath(uri, this.cordova);
+
+        // Get filename from uri
+        String fileName = realPath != null ? realPath.substring(realPath.lastIndexOf('/') + 1) : "modified.jpg";
```

Comment:
This doesn't respect `encodingType` parameter. In case if `encodingType == PNG` resultant filename still might have '.jpg' extension.